sky/engine/tonic/dart_message_handler.cc

```diff
@@ -0,0 +1,108 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "sky/engine/tonic/dart_message_handler.h"
+
+#include "base/bind.h"
+#include "base/logging.h"
+#include "dart/runtime/include/dart_api.h"
+#include "dart/runtime/include/dart_native_api.h"
+#include "dart/runtime/include/dart_tools_api.h"
+#include "sky/engine/tonic/dart_error.h"
+#include "sky/engine/tonic/dart_state.h"
+
+namespace blink {
+
+DartMessageHandler::DartMessageHandler()
+    : handled_first_message_(false),
+      task_runner_(nullptr) {
+}
+
+DartMessageHandler::~DartMessageHandler() {
+  task_runner_ = nullptr;
+}
+
+void DartMessageHandler::Initialize(
+    const scoped_refptr<base::SingleThreadTaskRunner>& runner) {
+  // Only can be called once.
+  CHECK(!task_runner_);
+  task_runner_ = runner;
+  Dart_SetMessageNotifyCallback(MessageNotifyCallback);
+}
+
+void DartMessageHandler::OnMessage(DartState* dart_state) {
+  CHECK(dart_state->message_handler().task_runner());
```

Comment:
Given that this is a CHECK rather than DCHECK, I'd probably grab this into a local variable and then CHECK it and use it.